MapboxCoreNavigation/RouteController.swift

```diff
@@ -216,8 +263,8 @@ extension RouteController: CLLocationManagerDelegate {
             
             if routeProgress.currentLegProgress.upComingStep?.maneuverType == ManeuverType.arrive {
                 alertLevel = .arrive
-            } else if courseMatchesManeuverFinalHeading {
-                routeProgress.currentLegProgress.stepIndex += 1
+            } else if courseMatchesManeuverFinalHeading && false {
```

Comment:
This is initentially false so we always fallback to the code that is implemented in this PR. Will revert this once we're good with the code.